tests/testthat/test_as_survey_twophase.r

```diff
@@ -16,37 +16,34 @@ d2pbc_survey <- twophase(id=list(~id,~id), data=pbc, subset=~randomized)
 d2pbc_srvyr <- pbc %>%
   as_survey_twophase(id = list(id, id), subset = randomized)
 
-
 survey_results <- list(
   as.data.frame(svymean(~bili, d2pbc_survey)),
   as.data.frame(svytotal(~bili, d2pbc_survey)),
   as.data.frame(svyquantile(~bili, d2pbc_survey, quantiles = 0.5)),
   data.frame(albumin = svyratio(~bili, ~albumin, d2pbc_survey)[[1]],
              se = sqrt(unname(svyratio(~bili, ~albumin, d2pbc_survey)[[2]])))
 ) %>% dplyr::bind_cols() %>%
-  setNames(c("mean", "mean_se", "total", "total_se", "median_q50", "ratio",
+  setNames(c("mean", "mean_se", "total", "total_se", "median", "ratio",
              "ratio_se")) %>%
   dplyr::tbl_df()
 
 srvyr_results <- d2pbc_srvyr %>%
   summarize(mean = survey_mean(bili),
             total = survey_total(bili),
-            median = survey_median(bili),
+            median = survey_median(bili, vartype = NULL),
```

Comment:
Can you add a test that checks whether `vartype = NULL` works across all functions (mean/total/median/var/ratio)?

Reply:
I added such tests and also generally tried do tidy a little _test-survey_statistics_basic.r_ and _test_survey_statistic.r_.

Moreover I added tests to cover assertion rules in functions computing survey statistics. This resulted also in adding few assertions to functions `survey_ratio` and `survey_quantile` to make them behave similarly to the others.

Lastly I modified _test_database.R_ a little, so that it won't throw warnings when it is run on a setup where databases are unavailable.